src/devtools/client/inspector/markup/actions/markup.js

```diff
@@ -106,9 +123,17 @@ export function selectionChanged(selection, expandSelectedNode) {
 
     // expand each ancestor, loading its children if necessary
     for (const ancestor of ancestors) {
+      if (shouldScrollIntoView) {
+        dispatch(scrollIntoView(ancestor.objectId()));
+      }
```

Comment:
I am kinda curious if we need this since we are scrolling to the selectedNode at the end.

Reply:
When selecting a node with the node picker, it can take some time until all ancestors are loaded and expanded - in that case it's nice to scroll the node that is currently loading into view. That's why I added `scrollIntoViewNode` because otherwise we could indeed just use `selectedNode`.